tests/unit_tests/scheduler/test_job.py

```diff
@@ -153,3 +158,101 @@ async def test_job_run_sends_expected_events(
         runpath=Path(realization.run_arg.runpath),
     )
     assert scheduler.driver.submit.call_count == max_submit
+
+
+@pytest.mark.asyncio
+async def test_job_run_logs_disk_sync_issue(realization: Realization, monkeypatch):
+    scheduler = create_scheduler()
+    scheduler.wait_for_checksum = lambda: True
+    file_path = "does/not/exist"
+    scheduler.checksum = {
+        "test_runpath": {
+            "file": {
+                "path": file_path,
+                "md5sum": "something",
+            }
+        }
+    }
+    log_msgs = []
+    job = Job(scheduler, realization)
+    job._verify_checksum = partial(job._verify_checksum, timeout=0)
+    job.started.set()
+
+    with captured_logs(log_msgs, logging.ERROR):
+        job_run_task = asyncio.create_task(job.run(asyncio.Semaphore(), max_submit=1))
+        job.started.set()
+        job.returncode.set_result(0)
+        await job_run_task
+
+    assert "Disk syncronization failed for does/not/exist" in log_msgs
+
+    await assert_scheduler_events(
+        scheduler, [State.SUBMITTING, State.PENDING, State.RUNNING, State.COMPLETED]
+    )
+
+
+@pytest.mark.asyncio
+async def test_job_run_logs_manifest_errors(realization: Realization, monkeypatch):
+    scheduler = create_scheduler()
+    scheduler.wait_for_checksum = lambda: True
+    file_path = "does/not/exist"
+    error = f"Expected file {file_path} not created by forward model!"
+    scheduler.checksum = {
+        "test_runpath": {
+            "file": {
+                "path": file_path,
+                "error": error,
+            }
+        }
+    }
+    log_msgs = []
+    job = Job(scheduler, realization)
+    job._verify_checksum = partial(job._verify_checksum, timeout=0)
+    job.started.set()
+
+    with captured_logs(log_msgs, logging.ERROR):
+        job_run_task = asyncio.create_task(job.run(asyncio.Semaphore(), max_submit=1))
+        job.started.set()
+        job.returncode.set_result(0)
+        await job_run_task
+
+    assert error in log_msgs
+
+    await assert_scheduler_events(
+        scheduler, [State.SUBMITTING, State.PENDING, State.RUNNING, State.COMPLETED]
```

Comment:
Shouldn't this be a `State.FAILED`. What do you think @xjules ?

Reply:
`forward_model_ok` is mocked in this tests hence the `State.Completed` final event. If it was not mocked then it would log another error when the file is not available. I can just remove the checking for the states that is not the point of this test.